pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -299,6 +303,31 @@ func (r *SubnetPortReconciler) SetupFieldIndexers(mgr ctrl.Manager) error {
 	return nil
 }
 
+func (r *SubnetPortReconciler) networkInfoMapFunc(ctx context.Context, networkInfo client.Object) []reconcile.Request {
+	subnetPortList := &v1alpha1.SubnetPortList{}
+	var requests []reconcile.Request
+	err := retry.OnError(retry.DefaultRetry, func(err error) bool {
+		return err != nil
+	}, func() error {
+		err := r.Client.List(ctx, subnetPortList, &client.ListOptions{Namespace: networkInfo.GetNamespace()})
+		return err
+	})
+	if err != nil {
+		log.Error(err, "failed to list SubnetPort", "Namespace", networkInfo.GetNamespace())
+		return requests
+	}
+	// Requeue the SubnetPort if the precreated VPC SNAT/LB status is changed
```

Comment:
why we need requeue SubnetPort if VPC NAT or LB changed?

Reply:
As discussed on chat, it's the privateTGW/external IPBlocks determine whether we need to reconcile the Subnet/SubnetPort/IPAddressBinding. We prefer to leave the error message to users and let them decide what to do after they modify the pre-created VPC.